Data/Aeson/Types/ToJSON.hs

```diff
@@ -257,15 +257,43 @@ genericLiftToEncoding opts te tel = gToEncoding opts (To1Args te tel) . from1
 -- than directly emitting an 'Encoding'.  Our one-liner definition of
 -- 'toEncoding' above bypasses the intermediate 'Value'.
 --
--- If @DefaultSignatures@ doesn't give exactly the results you want,
+-- If the default implementation doesn't give exactly the results you want,
 -- you can customize the generic encoding with only a tiny amount of
 -- effort, using 'genericToJSON' and 'genericToEncoding' with your
--- preferred 'Options':
+-- preferred 'Options'.
+--
+-- The following examples ensure that 'toEncoding' represents the same
+-- encoding as 'toJSON'.
+--
+-- - A typical implementation defines both 'toJSON' and 'toEncoding'
+--   with the same options.
+--
+-- @
+-- customOptions = 'defaultOptions'
+--                 { 'fieldLabelModifier' = 'map' 'Data.Char.toUpper'
+--                 }
+--
+-- instance 'ToJSON' Coord where
+--     'toJSON'     = 'genericToJSON' customOptions
+--     'toEncoding' = 'genericToEncoding' customOptions
+-- @
+--
+-- - A minimal definition of 'ToJSON' is given by 'toJSON'.
+--   This is most useful to define an instance manually when
+--   the generic options are not satisfactory.
+--
+-- @
+-- instance 'ToJSON' Coord where
+--      'toJSON' (Coord x y) = 'Array' (toJSON [x, y])
+-- @
+--
+-- - The default implementation of 'toJSON' uses 'defaultOptions'. If you leave
+--   'toJSON' implicit while providing a more efficient explicit implementation
+--   of 'toEncoding', then it *must* be equivalent to using 'defaultOptions'.
```

Comment:
This adds a second example for a manually written instance when we already have an example of this above. I'd prefer to swap/reword things a bit. I'll make a suggestions as a commit, one sec.